Keas.Mvc/Models/ExpiringItemsViewModel.cs

```diff
@@ -55,9 +57,70 @@ public static async Task<ExpiringItemsViewModel> Create(ApplicationDbContext con
+                a.Access.Team.Slug == teamName && !a.IsConfirmed && (a.Access.Active || a.Access.Active == !showInactive))
+                .Include(a => a.Access).Include(a=> a.Person).AsNoTracking().ToArrayAsync();
+            var expiringKey = await context.KeySerials.IgnoreQueryFilters().Where(a => (showType == "All" || showType == "Key") &&
+                (userRoles.Where(r => r.Name == Role.Codes.DepartmentalAdmin).Any() || userRoles.Where(r => r.Name == Role.Codes.KeyMaster).Any() || userRoles.Where(r => r.Name == Role.Codes.Admin).Any()) &&
+                a.Key.Team.Slug == teamName && !a.KeySerialAssignment.IsConfirmed && (a.Key.Active || a.Key.Active == !showInactive))
+                .Include(k => k.KeySerialAssignment).ThenInclude(a=> a.Person).Include(k => k.Key).AsNoTracking().ToArrayAsync();
+            var expiringEquipment = await context.Equipment.IgnoreQueryFilters().Where(a => (showType == "All" || showType == "Equipment") &&
+                 (userRoles.Where(r => r.Name == Role.Codes.DepartmentalAdmin).Any() || userRoles.Where(r => r.Name == Role.Codes.EquipmentMaster).Any() || userRoles.Where(r => r.Name == Role.Codes.Admin).Any()) &&
+                  a.Team.Slug == teamName && !a.Assignment.IsConfirmed && (a.Active || a.Active == !showInactive))
+                .Include(e => e.Assignment).ThenInclude(a=> a.Person).AsNoTracking().ToArrayAsync();
+            var expiringWorkstations = await context.Workstations.IgnoreQueryFilters().Where(a => (showType == "All" || showType == "Workstation") &&
+                (userRoles.Where(r => r.Name == Role.Codes.DepartmentalAdmin).Any() || userRoles.Where(r => r.Name == Role.Codes.SpaceMaster).Any() || userRoles.Where(r => r.Name == Role.Codes.Admin).Any()) &&
+                    a.Team.Slug == teamName && !a.Assignment.IsConfirmed && (a.Active || a.Active == !showInactive))
+                .Include(w => w.Assignment).ThenInclude(a=> a.Person).AsNoTracking().ToArrayAsync();
+
+            var itemList = populateItemList(userRoles);
+            var viewModel = new ReportItemsViewModel
+            {
+                Access = expiringAccess,
+                Keys = expiringKey,
+                Equipment = expiringEquipment,
+                Workstations = expiringWorkstations,
+                ShowInactive = showInactive,
+                ItemList =  itemList,
+                ShowType = showType                
+            };
+            return viewModel;
+        }
+
+        public static List<string> populateItemList(List<Role> userRoles) 
+        {
+            var itemList = new List<string>() {"All"};
+
+            if(userRoles.Where(r => r.Name == Role.Codes.DepartmentalAdmin).Any() || userRoles.Where(r => r.Name == Role.Codes.AccessMaster).Any() || userRoles.Where(r => r.Name == Role.Codes.Admin).Any())
```

Comment:
Maybe throw this into the security service?  This is at least 8 times in this one file where the exact same check is made.  It'll be hell to change if we ever add a new role

Reply:
:thumbsup: